src/stdlib/strings.bpf.c

```diff
@@ -218,8 +218,780 @@ m_str* __strerror(int errno, m_arg *out) {
+  [229] = "munlock",
+  [230] = "mlockall",
+  [231] = "munlockall",
+  [232] = "mincore",
+  [233] = "madvise",
+  [234] = "remap_file_pages",
+  [235] = "mbind",
+  [236] = "get_mempolicy",
+  [237] = "set_mempolicy",
+  [238] = "migrate_pages",
+  [239] = "move_pages",
+  [240] = "rt_tgsigqueueinfo",
+  [241] = "perf_event_open",
+  [242] = "accept4",
+  [243] = "recvmmsg",
+  [244] = "arch_specific_syscall",
+#if defined(__riscv)
+  [258] = "riscv_hwprobe",
+  [259] = "riscv_flush_icache",
+#endif
+  [260] = "wait4",
+  [261] = "prlimit64",
+  [262] = "fanotify_init",
+  [263] = "fanotify_mark",
+  [264] = "name_to_handle_at",
+  [265] = "open_by_handle_at",
+  [266] = "clock_adjtime",
+  [267] = "syncfs",
+  [268] = "setns",
+  [269] = "sendmmsg",
+  [270] = "process_vm_readv",
+  [271] = "process_vm_writev",
+  [272] = "kcmp",
+  [273] = "finit_module",
+  [274] = "sched_setattr",
+  [275] = "sched_getattr",
+  [276] = "renameat2",
+  [277] = "seccomp",
+  [278] = "getrandom",
+  [279] = "memfd_create",
+  [280] = "bpf",
+  [281] = "execveat",
+  [282] = "userfaultfd",
+  [283] = "membarrier",
+  [284] = "mlock2",
+  [285] = "copy_file_range",
+  [286] = "preadv2",
+  [287] = "pwritev2",
+  [288] = "pkey_mprotect",
+  [289] = "pkey_alloc",
+  [290] = "pkey_free",
+  [291] = "statx",
+  [292] = "io_pgetevents",
+  [293] = "rseq",
+  [294] = "kexec_file_load",
+  [424] = "pidfd_send_signal",
+  [425] = "io_uring_setup",
+  [426] = "io_uring_enter",
+  [427] = "io_uring_register",
+  [428] = "open_tree",
+  [429] = "move_mount",
+  [430] = "fsopen",
+  [431] = "fsconfig",
+  [432] = "fsmount",
+  [433] = "fspick",
+  [434] = "pidfd_open",
+  [435] = "clone3",
+  [436] = "close_range",
+  [437] = "openat2",
+  [438] = "pidfd_getfd",
+  [439] = "faccessat2",
+  [440] = "process_madvise",
+  [441] = "epoll_pwait2",
+  [442] = "mount_setattr",
+  [443] = "quotactl_fd",
+  [444] = "landlock_create_ruleset",
+  [445] = "landlock_add_rule",
+  [446] = "landlock_restrict_self",
+  [447] = "memfd_secret",
+  [448] = "process_mrelease",
+  [449] = "futex_waitv",
+  [450] = "set_mempolicy_home_node",
+  [451] = "cachestat",
+  [452] = "fchmodat2",
+  [453] = "map_shadow_stack",
+  [454] = "futex_wake",
+  [455] = "futex_wait",
+  [456] = "futex_requeue",
+};
+size_t syscall_names_size = sizeof(syscall_names)/sizeof(char*);
+#else
+static m_str syscall_names[1] = {};
+size_t syscall_names_size = 0;
+#endif
+
+static m_str unknown_syscall = "Unknown syscall";
+
+m_str* __syscall_name(unsigned n, m_arg *out) {
+  m_str *result;
+  char ch;
+
+  if (n < syscall_names_size) {
+    result = &syscall_names[n];
+  } else {
+    result = &unknown_syscall;
+  }
+
+  // Syscall numbers are not spaced by 1.
+  if (bpf_probe_read_kernel(&ch, sizeof(ch), (void*)*result) == 0) {
+    if (ch == '\0') {
+      result = &unknown_syscall;
+    }
+  } else {
+    result = &unknown_syscall;
+  }
+
+  bpf_probe_read_kernel(&out->data, sizeof(out->data), result);
```

Comment:
Per above, I think you just want to do the memcpy here, e.g. `  __builtin_memcpy`. See the strerror case above.